src/core/ready.js

```diff
@@ -49,11 +47,28 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !readyFiring ) {
+				readyFiring = true;
+				while ( readyCallbacks.length ) {
+					fn = readyCallbacks.shift();
+					if ( jQuery.isFunction( fn ) ) {
+						fn.call( document, jQuery );
```

Comment:
Because the invocations are synchronous and not wrapped in a `try` block, `jQuery.ready` is not a conforming [Promises/A+](https://promisesaplus.com/) promise. This should either include the explanatory comment from https://github.com/jquery/jquery/issues/1778#issuecomment-72764833 or be corrected (not that _that_ would be sufficient, but it would prevent callback lockups).